elide-datastore/elide-datastore-hibernate3/src/main/java/com/yahoo/elide/datastores/hibernate3/HibernateTransaction.java

```diff
@@ -99,7 +99,10 @@ public void delete(Object object) {
 
     @Override
     public void save(Object object) {
-        deferredTasks.add(() -> session.saveOrUpdate(object));
+        deferredTasks.add(() -> {
+            session.saveOrUpdate(object);
+            session.evict(object);
```

Comment:
I realized after I started commenting, this is some scratch work we were doing in-person on Asif's laptop last Friday. He submitted the test case for me so we could better debug and fix offline :)